lib/Analysis/Allocation.cpp

```diff
@@ -94,22 +88,33 @@ SmallVector<unsigned> getRepShapeForCvtLayout(RankedTensorType srcTy,
   return repShape;
 }
 
-SmallVector<unsigned>
-getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
-                             unsigned &outVec) {
-  auto srcTy = op.getSrc().getType();
-  auto dstTy = op.getType();
-  return getScratchConfigForCvtLayout(srcTy, dstTy, inVec, outVec);
+// TODO: extend beyond scalars
+static SmallVector<unsigned> getRepShapeForAtomicRMW(triton::AtomicRMWOp op) {
+  SmallVector<unsigned> smemShape;
+  if (isa<RankedTensorType>(op.getPtr().getType())) {
+    // do nothing or just assert because shared memory is not used in tensor up
```

Comment:
are we doing nothing, or are we asserting?

Reply:
The comments are from the old code, I just moved around